// @astrojs/db/dist/core/integration/vite-plugin-db-client.js
import { DB_CLIENTS, VIRTUAL_CLIENT_MODULE_ID } from "../consts.js";
function getRemoteClientModule(mode) {
  switch (mode) {
    case "web":
      return `export { createClient } from '${DB_CLIENTS.web}';`;
    case "node":
    default:
      return `export { createClient } from '${DB_CLIENTS.node}';`;
  }
}
function getLocalClientModule(mode) {
  switch (mode) {
    case "node":
    case "web":
    default:
      return `export { createClient } from '${DB_CLIENTS.local}';`;
  }
}
const resolved = "\0" + VIRTUAL_CLIENT_MODULE_ID;
function vitePluginDbClient(params) {
  return {
    name: "virtual:astro:db-client",
    enforce: "pre",
    async resolveId(id) {
      if (id !== VIRTUAL_CLIENT_MODULE_ID) return;
      return resolved;
    },
    async load(id) {
      if (id !== resolved) return;
      switch (params.connectToRemote) {
        case true:
          return getRemoteClientModule(params.mode);
        case false:
        default:
          return getLocalClientModule(params.mode);
      }
    }
  };
}
export {
  vitePluginDbClient
};
